ddclient

```diff
@@ -2307,6 +2306,24 @@ sub get_ip {
     return $ip;
 }
 
+######################################################################
+## is_ipv4() validates if string is valid IPv4 address
+######################################################################
+sub is_ipv4 {
+    my ($value) = @_;
+    return $value =~ /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\z/ai;
+}
+
+######################################################################
+## ipv4_match() extracts the first valid IPv4 address from given string
+######################################################################
+sub ipv4_match {
+    my ($value) = @_;
+    my $ip = undef;
+    $ip = $1 if $value =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
```

Comment:
This should reuse ``is_ipv4``.

Reply:
It can't.  At least I don't think it can.  The functions do two different things.  `is_ipv4` returns true/false and tests that a string exactly equals a valid IPv4 address... no text preceding or after the IPv4 in the string being tested.   `ipv4_match` on the other hand extracts the first valid IPv4 address it finds within a string (which could be multi-line, like output from `ifconfig`).  I'm not sure I like the name of the function but I chose it for consistency with existing `ipv6_match` function.

`ipv4_match` could be used instead of `is_ipv4` depending on what you want to do, but I don't think you can implement one function by calling the other.

Reply:
You could do it the other way around... make `is_ipv4` use `ipv4_match`...
```
sub is_ipv4 {
    my ($value) = @_;
    return $value eq ipv4_match($value);
}
```
The downside is it adds a string compare, but the upside is that the regex is only in the one place.  I like this, What do you think @SuperSandro2000 

David

Reply:
But would have to test for non-empty string as well.

Reply:
I think it would be better if we have the regex only once. I am not to deep into perl style but some global const (``IPV4_REGEX``) could work in other languages.

I did a sloppy read and thought you would duplicated code. Sorry about that.

Reply:
Optimized I think the most efficient is...
```
sub is_ipv4 {
    my ($value) = @_;
    return (length($value) != 0) && ($value eq ipv4_match($value));
}
```
So test for empty string before going into the regex match.

Reply:
> I did a sloppy read and thought you would duplicated code. Sorry about that.

No need to apologize... you prompted me to think about it and come up with a better way.  Maybe not more efficient, but definitely more readable.  Now I want to do the same to the IPv6 versions (as a separate PR after this one is merged)